variables.tf

```diff
@@ -34,16 +34,26 @@ variable "allow_credentials" {
   description = "Boolean value to allow authentication credentials."
   type        = bool
   default     = true
+
+  validation {
+    condition     = var.enable_cors == false && var.allow_credentials != null ? false : true
+    error_message = "var.allow_credentials can only be set if var.enable_cors is true"
+  }
 }
 
 variable "origins" {
   description = "An array of strings that contain allowed origin domains. "
   type        = list(string)
   default     = []
+
+  validation {
+    condition     = var.enable_cors == false && var.origins != null ? false : true
```

Comment:
This is not good logic. You are only checking for null, but the default value is `[]` meaning this will always fail if `enable_cors` is false (which it is by default).

Instead I suggest updating the logic in main.tf to only pass values to the `cors_config` block if `enable_cors` is true. That way you can keep the default value of `origins` as an empty list.

You should also update the variable description to say the value is only used if `enable_cors` is set to true